### PR TITLE
pgx Version 4.12.0 test and validation script

### PR DESCRIPTION
**Pull request by ambujkumar3**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

**Comment by seth-priya** [comment]
@prashantkhoje please review

File: p/pgx/pgx_ubi_8.5.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+PACKAGE_NAME=pgx
+PACKAGE_VERSION=v4.12.0
+PACKAGE_URL=https://github.com/jackc/pgx.git
+
+dnf install wget git -y
+
+PACKAGE_VERSION=${1:-v4.12.0}
+
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:31]:
```suggestion

PACKAGE_NAME=pgx
PACKAGE_VERSION=${1:-v4.12.0}
PACKAGE_URL=https://github.com/jackc/pgx.git

dnf install wget git -y
```

File: p/pgx/pgx_ubi_8.5.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+gpgcheck=0
+EOF
+
+yum group install -y 'Development Tools'
+yum install -y readline-devel
+
+if [ -d "postgres" ] ; then
+  rm -rf postgres
+
+fi
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:57]:
```suggestion
if [ -d "postgres" ] ; then
    rm -rf postgres
fi
```

File: p/pgx/pgx_ubi_8.5.sh [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+make
+make install
+cd ../../
+
+adduser postgres || true
+
+if [ -d "/usr/local/pgsql/data2" ] ; then
+  rm -rf /usr/local/pgsql/data2
+
+fi
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:76]:
```suggestion
if [ -d "/usr/local/pgsql/data2" ] ; then
    rm -rf /usr/local/pgsql/data2
fi
```

File: p/pgx/pgx_ubi_8.5.sh [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"
+
+fi
+
+if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:112]:
```suggestion
# Check if package exists
if [ -d "$PACKAGE_NAME" ] ; then
    rm -rf $PACKAGE_NAME
    echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"
fi

if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 0
fi
```

File: p/pgx/pgx_ubi_8.5.sh [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+#cd  $PACKAGE_NAME
+git checkout $PACKAGE_VERSION
+
+
+if ! (go build); then
+       echo "------------------$PACKAGE_NAME:build failed---------------------"
+       echo "$PACKAGE_VERSION $PACKAGE_NAME"
+       echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  build_Fails"
+       exit 1
+fi
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:123]:
```suggestion
cd /home/tester/$PACKAGE_NAME
git checkout $PACKAGE_VERSION

if ! (go build); then
    echo "------------------$PACKAGE_NAME:build failed---------------------"
    echo "$PACKAGE_VERSION $PACKAGE_NAME"
    echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  build_Fails"
    exit 1
fi
```

File: p/pgx/pgx_ubi_8.5.sh [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME" > /home/tester/output/test_fails
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails" > /home/tester/output/version_tracker
+        exit 1
+else
+        echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
+        echo "$PACKAGE_VERSION $PACKAGE_NAME" > /home/tester/output/test_success
+        echo "$PACKAGE_NAME  |  $PACKAGE_VERSION | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success" > /home/tester/output/version_tracker
+        exit 0
+fi
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:148]:
```suggestion
if ! PGX_TEST_DATABASE="host=127.0.0.1 port=5434 database=pgx_test user=pgx_pw password=secret sslmode=disable" go test -v ./...; then
    echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME" > /home/tester/output/test_fails
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails" > /home/tester/output/version_tracker
    exit 1
else
    echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
    echo "$PACKAGE_VERSION $PACKAGE_NAME" > /home/tester/output/test_success
    echo "$PACKAGE_NAME  |  $PACKAGE_VERSION | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success" > /home/tester/output/version_tracker
    exit 0
fi
```

**Review by prashantkhoje — CHANGES_REQUESTED** [review]
Minor formatting changes. Otherwise looks good.

File: p/pgx/pgx_ubi_8.5.sh [hunk 7]
```diff
@@ -0,0 +1,148 @@
+#!/bin/bash -e
```
Comment (prashantkhoje) [inline p/pgx/pgx_ubi_8.5.sh:1]:
```suggestion
#!/bin/bash -ex
```

**Comment by prashantkhoje** [comment]
This PR can be merged

**Pull request merged**